lib/experimental/class-wp-navigation-fallbacks-gutenberg.php

```diff
@@ -0,0 +1,241 @@
+<?php
+/**
+ * WP_Navigation_Fallbacks_Gutenberg class
+ *
+ * Manages fallback behavior for Navigation menus.
+ *
+ * @package Gutenberg
+ * @subpackage Navigation
+ * @since 6.3.0
+ */
+
+/**
+ * Import dependencies.
+ */
+require __DIR__ . '/class-wp-nav-menu-conversion-gutenberg.php';
+
+/**
+ * Manages fallback behavior for Navigation menus.
+ *
+ * @access public
+ */
+class WP_Navigation_Fallbacks_Gutenberg {
+
+	/**
+	 * Gets (and/or creates) an appropriate fallback Navigation Menu.
+	 *
+	 * @return WP_Post|null the fallback Navigation Post or null.
+	 */
+	public static function get_fallback_menu() {
+
+		$fallback = static::get_most_recently_published_navigation();
+
+		if ( $fallback ) {
+			return $fallback;
+		}
+
+		$fallback = static::create_classic_menu_fallback();
+
+		if ( $fallback && ! is_wp_error( $fallback ) ) {
+			return $fallback instanceof WP_Post ? $fallback : static::get_most_recently_published_navigation();
+		}
+
+		$fallback = static::create_default_fallback();
+
+		if ( $fallback && ! is_wp_error( $fallback ) ) {
+			return $fallback instanceof WP_Post ? $fallback : static::get_most_recently_published_navigation();
+		}
+
+		return null;
+	}
+
+	/**
+	 * Finds the most recently published `wp_navigation` Post.
+	 *
+	 * @return WP_Post|null the first non-empty Navigation or null.
+	 */
+	private static function get_most_recently_published_navigation() {
+
+		// Default to the most recently created menu.
+		$parsed_args = array(
+			'post_type'              => 'wp_navigation',
+			'no_found_rows'          => true,
+			'update_post_meta_cache' => false,
+			'update_post_term_cache' => false,
+			'order'                  => 'DESC',
+			'orderby'                => 'date',
+			'post_status'            => 'publish',
+			'posts_per_page'         => 1, // get only the most recent.
+		);
+
+		$navigation_post = new WP_Query( $parsed_args );
+
+		if ( count( $navigation_post->posts ) > 0 ) {
+			return $navigation_post->posts[0];
+		}
+
+		return null;
+	}
+
+	/**
+	 * Creates a Navigation Menu post from a Classic Menu.
+	 *
+	 * @return int|WP_Error The post ID of the default fallback menu or a WP_Error object.
+	 */
+	private static function create_classic_menu_fallback() {
+		// See if we have a classic menu.
+		$classic_nav_menu = static::get_fallback_classic_menu();
+
+		if ( ! $classic_nav_menu ) {
+			return new WP_Error( 'no_classic_menus', __( 'No Classic Menus found.', 'gutenberg' ) );
+		}
+
+		// Todo: inject as dependency.
+		$menu_converter = new WP_Nav_Menu_Conversion_Gutenberg( $classic_nav_menu );
+
+		// If we have a classic menu then convert it to blocks.
+		$classic_nav_menu_blocks = $menu_converter->convert();
```

Comment:
Unsure if the converter should be instantiated in the light of [this comment](https://github.com/WordPress/gutenberg/pull/48698/files#r1171489996). See my reconsideration [here](https://github.com/WordPress/gutenberg/pull/48698/files#diff-b8977663d15d4fca5de662b8a4bcf61ed55387d25cec34d1bc28c3e17a47805cR28).

Reply:
Now updated to static.